packages/dev/codemods/src/s1-to-s2/src/codemods/components/ActionGroup/transform.ts

```diff
@@ -0,0 +1,187 @@
+import {addComponentImport} from '../../shared/utils';
+import {commentOutProp} from '../../shared/transforms';
+import {getComponents} from '../../../getComponents';
+import {NodePath} from '@babel/traverse';
+import * as t from '@babel/types';
+
+let availableComponents = getComponents();
+
+/**
+ * Transforms ActionGroup:
+ * - Comment out overflowMode (it has not been implemented yet).
+ * - Comment out buttonLabelBehavior (it has not been implemented yet).
+ * - Comment out summaryIcon (it has not been implemented yet).
+ * - Use ActionButtonGroup if no selection.
+ * - Use ToggleButtonGroup if selection is used.
+ * - Update root level onAction to onPress on each ActionButton.
+ * - Apply isDisabled directly on each ActionButton/ToggleButton instead of disabledKeys.
+ * - Update key to id (keep key for map).
+ * - Convert dynamic collections render function to items.map.
+ */
+export default function transformActionGroup(path: NodePath<t.JSXElement>) {
+  // Comment out overflowMode
+  commentOutProp(path, {propToComment: 'overflowMode'});
+
+  // Comment out buttonLabelBehavior
+  commentOutProp(path, {propToComment: 'buttonLabelBehavior'});
+
+  // Comment out summaryIcon
+  commentOutProp(path, {propToComment: 'summaryIcon'});
+
+  let selectionModePath = path.get('openingElement').get('attributes').find((attr) => t.isJSXAttribute(attr.node) && attr.node.name.name === 'selectionMode') as NodePath<t.JSXAttribute> | undefined;
```

Comment:
everything below this was the choice of rendering an ActionGroup vs a ToggleGroup?  maybe move it out to a function named like it was in the declarative case so it's easier to review. I don't know if there is new logic in this transform that I should be looking at

same goes for other components I assume

Reply:
```
      {
        description: 'Replace with ActionButtonGroup or ToggleButtonGroup',
        reason: 'The API has changed',
        function: {name: 'updateActionGroup', args: {}}
      }
```

I assume everything below is what used to be `updateActionGroup`

Reply:
> I assume everything below is what used to be `updateActionGroup`

Exactly. I can clean this up a bit.